subsys/debug/thread_analyzer.c

```diff
@@ -37,19 +37,19 @@ LOG_MODULE_REGISTER(thread_analyzer, CONFIG_THREAD_ANALYZER_LOG_LEVEL);
 
 static void thread_print_cb(struct thread_analyzer_info *info)
 {
-	unsigned int pcnt = (info->stack_used * 100U) / info->stack_size;
+	uint32_t pcnt = (info->stack_used * 100U) / info->stack_size;
```

Comment:
`size_t` seems to be more correct in that case. stack_used and stack_size are both that type. And then `zu` below should remain.

Reply:
the percentage result does not represent a size anymore. The issue we are trying to fix here however is when building for 64bit platforms... 
```
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:44:4: warning: format '%zu' expects argument of type 'size_t', but argument 6 has type 'unsigned int' [-Wformat=]
   44 |    " %-20s: STACK: unused %zu usage %zu / %zu (%zu %%); CPU: %zu %%"),
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:21:43: note: in definition of macro 'THREAD_ANALYZER_PRINT'
   21 | #define THREAD_ANALYZER_PRINT(...) printk(__VA_ARGS__)
      |                                           ^~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:43:3: note: in expansion of macro 'THREAD_ANALYZER_FMT'
   43 |   THREAD_ANALYZER_FMT(
      |   ^~~~~~~~~~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:44:50: note: format string is defined here
   44 |    " %-20s: STACK: unused %zu usage %zu / %zu (%zu %%); CPU: %zu %%"),
      |                                                ~~^
      |                                                  |
      |                                                  long unsigned int
      |                                                %u
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:44:4: warning: format '%zu' expects argument of type 'size_t', but argument 7 has type 'unsigned int' [-Wformat=]
   44 |    " %-20s: STACK: unused %zu usage %zu / %zu (%zu %%); CPU: %zu %%"),
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:21:43: note: in definition of macro 'THREAD_ANALYZER_PRINT'
   21 | #define THREAD_ANALYZER_PRINT(...) printk(__VA_ARGS__)
      |                                           ^~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:43:3: note: in expansion of macro 'THREAD_ANALYZER_FMT'
   43 |   THREAD_ANALYZER_FMT(
      |   ^~~~~~~~~~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/subsys/debug/thread_analyzer.c:44:64: note: format string is defined here
   44 |    " %-20s: STACK: unused %zu usage %zu / %zu (%zu %%); CPU: %zu %%"),
      |                                                              ~~^
      |                                                                |
      |                                                                long unsigned int
      |                                                              %u
``

Reply:
@nashif Sorry, I mean `zu` should remain for the pcnt, changing for utilization makes sense, I missed that.
So I think this would be the correct ones:
`size_t pcnt = (info->stack_used * 100U) / info->stack_size;`
and
`" %-20s: STACK: unused %zu usage %zu / %zu (%zu %%); CPU: %u %%"),`

I checked and it compiles fine.